src/dsync/dsync.c

```diff
@@ -2869,6 +2871,91 @@ static int dsync_validate_link_dest(const char *link_dest, const char *dest, mfu
     }
 }
 
+#ifdef DAOS_SUPPORT
+/* Setup DAOS for dsync */
+static int dsync_daos_setup(
+    int rank,
+    daos_args_t* daos_args,
+    char** argpaths,
+    mfu_file_t* mfu_src_file,
+    mfu_file_t* mfu_dst_file)
+{
+    /* For error handling */
+    bool daos_do_cleanup = false;
+    bool daos_do_exit = false;
+
+    /* Set up DAOS arguments, containers, dfs, etc. */
+    int daos_rc = daos_setup(rank, argpaths, daos_args, mfu_src_file, mfu_dst_file);
+    if (daos_rc != 0) {
+        return 1;
+    }
+
+    /* DAOS does not support hard links */
+    if (options.link_dest != NULL &&
+            (mfu_src_file->type == DAOS || mfu_src_file->type == DFS ||
+             mfu_dst_file->type == DAOS || mfu_dst_file->type == DFS)) {
+        MFU_LOG(MFU_LOG_ERR, "DAOS does not support --link-dest.");
+        return 1;
+    }
+
+    /* Not yet supported */
+    if (options.delete && mfu_src_file->type == DAOS || mfu_dst_file->type == DAOS) {
+        MFU_LOG(MFU_LOG_ERR, "DAOS API does not support --delete.");
+        return 1;
+    }
+
+    return 0;
+}
+
+/* Perform a DAOS object sync */
+static int dsync_daos(
```

Comment:
@daltonbohning confused why other functions are named with mfu_daos_ but these all start with dsync_? Is there a reason for this?

Reply:
@dsikich Really just a question of namespaces. Since these functions are defined within `dsync.c` I thought it made sense to give them a `dsync_` prefix, particularly because they are only relevant in the context of `dsync`.
However, I'm open to criticism :)

Reply:
@daltonbohning was just curious since I don't think we have any functions that begin with dcp_ but it is just a helper function in dsync, so I don't think it is that big a deal since it is clear it is a daos function.